pandevice/base.py

```diff
@@ -1873,7 +1876,10 @@ def parse_xml(self, xml):
 
     def element_str(self):
         """The XML form of this object (and its children) as a string."""
-        return ET.tostring(self.element())
+        val = ET.tostring(self.element())
+        if hasattr(val, "hex"):
```

Comment:
XML on PAN-OS is utf8.  suggest to just use encoding='utf8' for 2 and 3.  2 will be str, 3 will be bytes.  3 needs bytes for urlopen() so unicode would need to be encoded back to bytes by pan.xapi.